tasks/git/src/main/java/com/walmartlabs/concord/plugins/git/GitHubTaskAction.java

```diff
@@ -0,0 +1,47 @@
+package com.walmartlabs.concord.plugins.git;
+
+import com.walmartlabs.concord.plugins.git.client.GitHubClient;
+import com.walmartlabs.concord.plugins.git.model.GitHubApiInfo;
+
+import java.util.Map;
+import java.util.UUID;
+
+public abstract class GitHubTaskAction<T extends GitHubTaskParams> {
+
+    public abstract Action action();
+
+    public abstract Map<String, Object> execute(UUID txId, GitHubApiInfo apiInfo, T input) throws Exception;
+
+    protected GitHubClient createClient(UUID txId, GitHubApiInfo apiInfo) {
```

Comment:
When we would want to override this method instead of just calling `new GitHubClient(txId, apiInfo)`?